tests/test_cli_prompters.cpp

```diff
@@ -51,16 +52,91 @@ TEST_F(CLIPrompters, PlainReturnsText)
     EXPECT_EQ(prompt.prompt(""), "value");
 }
 
-class CLIPromptersCinState : public CLIPrompters, public WithParamInterface<std::ios_base::iostate>
+// Note that the following test does not actually test if the terminal echos or not-
+// that is specific to platform terminal types.
+TEST_F(CLIPrompters, passphraseCallsEchoAndReturnsExpectedPassphrase)
+{
+    const std::string expected_output{"Please enter passphrase: \n"};
+    const std::string passphrase{"foo"};
+    mpt::MockTerminal mock_terminal;
+
+    cin.str(passphrase + "\n");
+
+    EXPECT_CALL(mock_terminal, set_cin_echo(false)).Times(1);
+    EXPECT_CALL(mock_terminal, set_cin_echo(true)).Times(1);
+    EXPECT_CALL(mock_terminal, cout()).WillRepeatedly([this]() -> std::ostream& { return cout; });
+    EXPECT_CALL(mock_terminal, cin()).WillOnce([this]() -> std::istream& { return cin; });
```

Comment:
Why not this?

```suggestion
    EXPECT_CALL(mock_terminal, cout()).WillRepeatedly(ReturnRef(cout));
    EXPECT_CALL(mock_terminal, cin()).WillOnce(ReturnRef(cin));
```

Below, too.

Reply:
Right, not sure why I missed that....

Reply:
Done